### ISSUE TITLE
Make vault name not unique

### ISSUE THREAD

**Issue by tobihagemann** [cryptomator/hub#147]
### Please agree to the following

- [X] I have searched [existing issues](https://github.com/cryptomator/hub/issues?q=) for duplicates
- [X] I agree to follow this project's [Code of Conduct](https://github.com/cryptomator/hub/blob/develop/.github/CODE_OF_CONDUCT.md)

### Summary

The name of a vault shouldn't be unique.

### Motivation

Feedback from a beta tester:

> The fact that the name of the vault must be globally unique would probably be problematic in a larger rollout.

### Anything else?

I'm not aware of any technical reason why the vault name has to be unique. I believe it was a decision in favor of usability when vaults didn't have a description. But the description should probably suffice to make vaults distinguishable. Of course, this would make #101 more important.

**Comment by overheadhunter** [comment]
Now, that #101 and #195 are solved, should we add this to 1.2.0?

**Comment by overheadhunter** [comment]
@tobihagemann looks good. However, this catch block can be removed altogether, since an uncaught `PersistenceException` will result in error 500 anyway:

```java
} catch (PersistenceException e) {
	throw new InternalServerErrorException(e);
}
```
